WordPress/gutenberg
# BaseControl: Add opt-in prop for margin-free styles

Part 1 of #38730
Prerequisite for #38720 

## What?

Adds a `__nextHasNoMarginBottom` prop so consumers can opt-in to the new styles without outer margins.

## Why?

As described in #38730, components that include outer margins are hard to reuse in different contexts.

## How?

Generally follows the [style deprecation procedure](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#deprecating-styles) in the docs, except we **aren't formally deprecating** the margin styles yet. We should initiate the formal deprecation (i.e. trigger a deprecation warning) only after all internal usages have been migrated.

## Testing Instructions

1. `npm run storybook:dev` and check out the stories for `BaseControl`.
2. When `__nextHasNoMarginBottom` is `true`, the component should have no outer bottom margin. (Regardless of whether it includes a `help` text.)
